examples/versioning/src/main.rs

```diff
@@ -68,3 +72,51 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use axum::{body::Body, http::Request, http::StatusCode};
+    use http_body_util::BodyExt;
+    use tower::ServiceExt;
+
+    #[tokio::test]
+    async fn test_v1() {
+        let response = app()
+            .oneshot(
+                Request::builder()
+                    .uri("/v1/foo")
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+        let body = response.into_body();
+        let bytes = body.collect().await.unwrap().to_bytes();
+        let html = String::from_utf8(bytes.to_vec()).unwrap();
+
+        assert_eq!(html, "received request with version V1");
+    }
+
+    #[tokio::test]
+    async fn test_v4() {
+        let response = app()
+            .oneshot(
+                Request::builder()
+                    .uri("/v4/foo")
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::NOT_FOUND);
```

Comment:
why is it not found?

Reply:
Because the enum has v1,v2, v3
I could use `"/blabla/foo"` in the example if you think that makes it clearer.